src/win/fs.c

```diff
@@ -1497,13 +1500,24 @@ static void fs__fchmod(uv_fs_t* req) {
 }
 
 
-INLINE static int fs__utime_handle(HANDLE handle, double atime, double mtime) {
-  FILETIME filetime_a, filetime_m;
+INLINE static int fs__utime_handle(HANDLE handle,
+                                   double btime,
+                                   double atime,
+                                   double mtime) {
+  FILETIME filetime_b, filetime_a, filetime_m;
+  int btime_provided = !isnan(btime);
+
+  if (btime_provided) {
```

Comment:
Instead of having `btime_provided`, could you make the `isnan()` call the conditional, and just assign `filetime_b` to a value or `NULL`. Then, you can also get rid of the ternary below.